istio/resources/values-template.yaml

```diff
@@ -1,445 +1,193 @@
+    }
+    rule {
+      action = "replace"
+      source_labels = ["__meta_kubernetes_pod_name"]
+      target_label  = "pod"
+    }
+  }
+
+  prometheus.scrape "istio_proxy_metrics" {
+    targets      = discovery.relabel.istio_proxy_metrics.output
+    forward_to   = [prometheus.relabel.metrics_service.receiver]
+    metrics_path = "/stats/prometheus"
+  }
+
+  discovery.relabel "istio_istiod_metrics" {
+    targets = discovery.kubernetes.endpoints.targets
+
+    rule {
+      action        = "keep"
+      source_labels = ["__meta_kubernetes_service_name", "__meta_kubernetes_endpoint_port_name"]
+      regex         = "istiod;http-monitoring"
+    }
+    rule {
+      source_labels = ["__meta_kubernetes_pod_annotation_prometheus_io_port", "__meta_kubernetes_pod_ip"]
+      regex         = "(\\d+);(([A-Fa-f0-9]{1,4}::?){1,7}[A-Fa-f0-9]{1,4})"
+      target_label  = "__address__"
+      replacement   = "[$2]:$1"
+    }
+    rule {
+      source_labels = ["__meta_kubernetes_pod_annotation_prometheus_io_port", "__meta_kubernetes_pod_ip"]
+      regex         = "(\\d+);((([0-9]+?)(\\.|$)){4})"
+      target_label  = "__address__"
+      replacement   = "$2:$1"
+    }
+    rule {
+      target_label = "job"
+      replacement = "integrations/istio"
+    }
+    rule {
+      target_label  = "cluster"
+      replacement   = "minikube"
+    }
+    rule {
+      source_labels = ["__meta_kubernetes_namespace", "__meta_kubernetes_pod_name"]
+      separator = "-"
+      target_label = "instance"
+    }
+    rule {
+      action = "replace"
+      source_labels = ["__meta_kubernetes_pod_name"]
+      target_label  = "pod"
+    }
+  }
+
+  prometheus.scrape "istio_istiod_metrics" {
+    targets    = discovery.relabel.istio_istiod_metrics.output
+    forward_to   = [prometheus.relabel.metrics_service.receiver]
+  }
+
+logs:
+  extraConfig: |-
+    discovery.relabel "istio_proxy_logs" {
+      targets = discovery.kubernetes.pods.targets
+
+      rule {
+        action        = "keep"
+        source_labels = ["__meta_kubernetes_pod_container_name"]
+        regex         = "istio-proxy.*"
       }
-
-      discovery.relabel "istiod_filter" {
-        targets = discovery.kubernetes.istiod.targets
-
-        rule {
-          source_labels = ["__meta_kubernetes_service_name", "__meta_kubernetes_endpoint_port_name"]
-          regex         = "istiod;http-monitoring"
-          action        = "keep"
-        }
-
-        rule {
-          action = "replace"
-          source_labels = ["__meta_kubernetes_pod_name"]
-          target_label  = "pod"
-        }
+      rule {
+        target_label = "job"
+        replacement = "integrations/istio"
       }
-
-      prometheus.scrape "istiod" {
-        targets    = discovery.relabel.istiod_filter.output
-        forward_to = [prometheus.remote_write.cloud.receiver]
-        job_name   = "integrations/istio"
+      rule {
+        target_label  = "cluster"
+        replacement   = "minikube"
```

Comment:
Did you find that you had to specifically set these in order for them to come through?

In the helm chart, if you're using the defined `loki.write` it should be adding the cluster as an [external label](https://github.com/grafana/k8s-monitoring-helm/blob/main/charts/k8s-monitoring/templates/agent_config/_loki.river.txt#L30-L35).

Reply:
@rgeyer nope! I actually removed them from my cloud-onboarding snippet but forgot to do so from here. I'll remove them in both spots.